crypto/asn1/i2d_evp.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 1995-2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* We need to use some deprecated APIs to support the legacy bits */
+#define OPENSSL_SUPPRESS_DEPRECATED
+
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/evp.h>
+#include <openssl/encoder.h>
+#include <openssl/buffer.h>
+#include <openssl/x509.h>
+#include <openssl/rsa.h>         /* For i2d_RSAPublicKey */
+#include <openssl/dsa.h>         /* For i2d_DSAPublicKey */
+#include <openssl/ec.h>          /* For i2o_ECPublicKey */
+#include "crypto/asn1.h"
+#include "crypto/evp.h"
+
+static int i2d_provided(OSSL_LIB_CTX *libctx,
+                        const EVP_PKEY *a, const char *propq,
+                        int selection, const char *output_structures[],
+                        unsigned char **pp)
+{
+    OSSL_ENCODER_CTX *ctx = NULL;
+    int ret = -1;
+    size_t len = INT_MAX;
```

Comment:
Maybe keep the comment that *pp is unbounded on input?

Reply:
Someone will have to propose the comment text.  My last attempt was incomprehensible to @slontis :wink:

Reply:
`The i2d_ API calls do not provide boundary length of *pp, we keep it unbounded.` or something like that?

Reply:
Thanks.  I rephrased it a bit more, but that was a good start.  If @slontis disapproves, I leave it to you two to resolve it, yeah?  :wink: